rclcpp/test/rclcpp/test_publisher.cpp

```diff
@@ -485,6 +495,11 @@ class TestPublisherProtectedMethods : public rclcpp::Publisher<MessageT, Allocat
 TEST_F(TestPublisher, do_loaned_message_publish_error) {
   initialize();
   using PublisherT = TestPublisherProtectedMethods<test_msgs::msg::Empty, std::allocator<void>>;
+  // This test only passes when message is allocated on heap, not middleware.
+  // Since `do_loaned_message_publish()` will fail, there is no way to return the message
+  // to the middleware.
+  // This eventually fails to destroy publisher handle in the implementation.
+  ASSERT_TRUE(rcutils_set_env("ROS_DISABLE_LOANED_MESSAGES", "1"));
```

Comment:
this test is intended to check the method if it generates the expected exception, which will not actually call the middleware at all.
so i believe that disabling `LoanedMessage` is okay to proceed the test here.